src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -91,7 +92,12 @@ public static SamInputResource of(final File file) {
 
     /** Creates a {@link SamInputResource} reading from the provided resource, with no index. */
     public static SamInputResource of(final Path path) {
-        return new SamInputResource(new PathInputResource(path));
+
+        if (Files.isRegularFile(path) &&  Files.exists(path)) {
+            return new SamInputResource(new PathInputResource(path));
+        } else {
```

Comment:
This needs a comment linking back to the discussion that spawned it.  Otherwise it will cause much confusion in the future.